sdk/scheduler/src/main/java/com/mesosphere/sdk/config/DefaultConfigurationUpdater.java

```diff
@@ -83,6 +78,31 @@ public UpdateResult updateConfiguration(ServiceSpec candidateConfig) throws Conf
                     String.format("Unable to serialize new config to JSON for logging: %s", e.getMessage())));
         }
 
+        if (targetConfig == null) {
+            LOGGER.info("Skipping config diff: There is no old config target to diff against");
+        } else if (candidateConfigJson == null) {
+            LOGGER.info("Skipping config diff: New target couldn't be represented as JSON");
```

Comment:
This could happen if candidateConfig.toJsonString() threw an exception (couple lines up). Error handling in that case is to add an error to the plan, but we wouldnt want that to be short-circuited by an NPE here.